src/Lazy/zipWith.ts

```diff
@@ -0,0 +1,61 @@
+import { map, zip } from ".";
+import { isAsyncIterable, isIterable } from "../_internal/utils";
+
+/**
+ * Returns Iterable/AsyncIterable out of the two supplied by applying `f` to each same positioned pair in Iterable/AsyncIterable.
+ *
+ * @example
+ * const iter = zipWith((a,b) => [a,b], [1,2,3], ['a','b','c']);
+ * iter.next(); // {value: [1, 'a'] , done: false}
+ * iter.next(); // {value: [2, 'b'] , done: false}
+ * iter.next(); // {value: [3, 'c'] , done: false}
+ * iter.next(); // {value: undefined , done: true}
+ */
+function zipWith<A, B, C>(
+  f: (a: A, b: B) => C,
+  iterable1: Iterable<A>,
+  iterable2: Iterable<B>,
+): IterableIterator<C>;
+
+function zipWith<A, B, C>(
+  f: (a: A, b: B) => C,
+  iterable1: Iterable<A>,
+  iterable2: AsyncIterable<B>,
+): AsyncIterableIterator<C>;
+
+function zipWith<A, B, C>(
+  f: (a: A, b: B) => C,
+  iterable1: AsyncIterable<A>,
+  iterable2: Iterable<B>,
+): AsyncIterableIterator<C>;
+
+function zipWith<A, B, C>(
+  f: (a: A, b: B) => C,
+  iterable1: AsyncIterable<A>,
+  iterable2: AsyncIterable<B>,
+): AsyncIterableIterator<C>;
+
+function zipWith<A, B, C>(
+  f: (a: A, b: B) => C,
+  iterable1: Iterable<A> | AsyncIterable<A>,
+  iterable2: Iterable<B> | AsyncIterable<B>,
+): IterableIterator<C> | AsyncIterableIterator<C> {
+  if (isIterable(iterable1) && isIterable(iterable2)) {
+    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
+  }
+  if (isIterable(iterable1) && isAsyncIterable(iterable2)) {
+    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
+  }
+  if (isAsyncIterable(iterable1) && isIterable(iterable2)) {
+    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
+  }
+  if (isAsyncIterable(iterable1) && isAsyncIterable(iterable2)) {
+    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
+  }
```

Comment:
The current code makes the intent a little clearer, but I think it's fine to keep it concise in this way.
```suggestion
  if (isAsyncIterable(iterable1)) {
    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
  }
  if (isAsyncIterable(iterable2)) {
    return map(([a, b]) => f(a, b), zip(iterable1, iterable2));
  }
```

Reply:
You are right. However, I think it is right to check TypeError in all situations because it is a code written to notify in runtime situations

Reply:
Okay, this is good enough.